language/dsl/src/__tests__/template.test.tsx

```diff
@@ -75,6 +75,114 @@ test("works if already in a template array", async () => {
   });
 });
 
+test("Template Order is preserved", async () => {
```

Comment:
The tests here cover cases where `<Template>` is added to the beginning or end of the children, but what about cases when the template is somewhere in the middle? Something like:
```ts
<Collection.Values>
  <Text>First Item</Text>
  <Template data={b`foo`}>
    <Text>Dynamic Item _index_</Text>
  </Template>
  <Text>Last Item</Text>
</Collection.Values>
```

I'm not sure this will work right now without changes to player core since you can't generate JSON like:
```json
{
  "id": "collection",
  "type": "collection",
  "values": [
    {/* first text asset */}
  ],
  "template": [
    {/* template here */}
  ],
  "values": [
    {/* last text asset */}
  ]
}
```

I feel like there needs to be a mechanism to tell player how it should resolve the template against existing data, like having the `insertionIndex` be an optional property of the actual `template` itself so it can insert its generated values correctly at runtime.

Reply:
@tmarmer What would happen with these changes is the template property would be inserted relative to the first encountered template in a slot. This really is a limitation of the current template API as, while ideal, that JSON wouldn't be valid so Player wouldn't be able to parse it. There is an [issue](https://github.com/player-ui/player/issues/541) in the Player repo to expand the API to support more complex insertion scenarios but this should be in parity with the current API

Reply:
Thanks Ketan, I wasn't aware that there was already an issue to extend this solution further.